### ISSUE TITLE
`unused-lifetimes` lint duplicates "parameter is never used" error

### ISSUE THREAD

**Issue by Aaron1011** [rust-lang/rust#72587]
The following code:

```rust
#![warn(unused_lifetimes)]
struct Foo<'a>;

fn main() {}

```

gives the following output:

```
arning: lifetime parameter `'a` never used
 --> unused_lifetime.rs:2:12
  |
2 | struct Foo<'a>;
  |           -^^- help: elide the unused lifetime
  |
note: the lint level is defined here
 --> unused_lifetime.rs:1:9
  |
1 | #![warn(unused_lifetimes)]
  |         ^^^^^^^^^^^^^^^^

error[E0392]: parameter `'a` is never used
 --> unused_lifetime.rs:2:12
  |
2 | struct Foo<'a>;
  |            ^^ unused parameter
  |
  = help: consider removing `'a`, referring to it in a field, or using a marker such as `std::marker::PhantomData`

error: aborting due to previous error; 1 warning emitted

For more information about this error, try `rustc --explain E0392`.
```

The "unused lifetimes" warning is redundant - not using a lifetime parameter is a hard error, so the warning gives no additional information.

**Comment by awoimbee** [comment]
Note that this lint still makes sense for functions:
```rust
#![warn(unused_lifetimes)]
fn unused_lifetime<'a>() {}
fn main() {unused_lifetime();}
```
```
   Compiling aaaaaa v0.1.0 (/tmp/aaaaaa)
warning: lifetime parameter `'a` never used
 --> src/main.rs:3:20
  |
3 | fn unused_lifetime<'a>() {}
  |                   -^^- help: elide the unused lifetime
  |
note: lint level defined here
 --> src/main.rs:1:9
  |
1 | #![warn(unused_lifetimes)]
  |         ^^^^^^^^^^^^^^^^

    Finished dev [unoptimized + debuginfo] target(s) in 0.16s
```
But then running clippy on the following looks dumb (the lints are redundant):
```rust
#![warn(unused_lifetimes)]
#![warn(clippy::extra_unused_lifetimes)]
fn unused_lifetime<'a>() {}
fn main() {
	unused_lifetime();
}
```
```
    Checking aaaaaa v0.1.0 (/tmp/aaaaaa)
warning: lifetime parameter `'a` never used
 --> src/main.rs:3:20
  |
3 | fn unused_lifetime<'a>() {}
  |                   -^^- help: elide the unused lifetime
  |
note: lint level defined here
 --> src/main.rs:1:9
  |
1 | #![warn(unused_lifetimes)]
  |         ^^^^^^^^^^^^^^^^

warning: this lifetime isn't used in the function definition
 --> src/main.rs:3:20
  |
3 | fn unused_lifetime<'a>() {}
  |                    ^^
  |
note: lint level defined here
 --> src/main.rs:2:9
  |
2 | #![warn(clippy::extra_unused_lifetimes)]
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#extra_unused_lifetimes

    Finished dev [unoptimized + debuginfo] target(s) in 0.10s
```

I don't really know the compiler internals, but should this lint just skip structs ? (is that even possible ?)

**Comment by oskgo** [comment]
@rustbot +E-needs-test

This was fixed in #96833 but no test was added for nonduplication.
